modules/sql/src/main/scala/gem/sql/FiniteDuration.scala

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.sql
+
+/** Minimal time classes to support reading from the enum
+  * tables.
+  */
+sealed trait FiniteDuration extends Product with Serializable {
+  def toMillis: Long
+
+}
+
+object FiniteDuration {
+  final case class Seconds(toMillis: Long) extends FiniteDuration
+  final case class Milliseconds(toMillis: Long) extends FiniteDuration
```

Comment:
Following the `Wavelength` example, it seems like these should be `BigDecimal` fields to hold the `numeric` values.  Then you could add a `toMilliseconds` in the `trait` that does the conversion.

```scala
sealed trait FiniteDuration extends Product with Serializable {
  def toBigDecimal: BigDecimal

  def toMilliseconds: Long = {
    val n = this match {
      case FiniteDuration.Milliseconds(_) => 0
      case FiniteDuration.Seconds(_)      => 3
    }
    toBigDecimal.underlying.scaleByPowerOfTen(n).setScale(0, HALF_UP).longValue
  }
}
...
```